### PR TITLE
Fix #323: Pull in workflow diagram

### LINKED ISSUE #323: Update workflow diagram
The coffee editor seems to use an outdated version of the workflow diagram example. Do we have plans to update it to the latest state including new features?

### PR DESCRIPTION
**Pull request by cdamus**
In order to facilitate demonstration of the Coffee Editor as an all-in-one example, and likewise to make it feasible to provide an all-in-one example of adding a new model concept to the editor in #315, this PR incorporates the Sprotty implementation of the workflow diagram into the Coffee Editor repository (it was currently is a binary dependency).

- incorporate the latest from `workflow-sprotty` on the
  `workflow` branch of the **eclipse-glsp-examples** repo
- adapt to removal of `commandExecutorModule` and
 ` glspEditLabelValidationModule` from GLSP

Signed-off-by: Christian W. Damus <give.a.damus@gmail.com>

### CODE REVIEW

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
```
Comment (eneufeld) [inline web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:2]: I don't know if the linter is configure wrong, but this should probably be 
`Copyright (c) 2021 EclipseSource and others.`
or 
`Copyright (c) 2020 - 2021 EclipseSource and others.`

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
```
Comment (eneufeld) [inline web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:14]: I think we use 
```suggestion
 * SPDX-License-Identifier: EPL-2.0 OR MIT
```

**Review by eneufeld — COMMENTED** [review]
Thank you for your commit, I have 3 high level remarks before I do a deeper dive into the code

File: web/coffee-workflow-glsp-editor/src/browser/diagram/workflow-diagram-configuration.ts [hunk 3]
```diff
@@ -27,6 +26,7 @@ import { Container, inject, injectable } from 'inversify';
 import { DiagramConfiguration, TheiaDiagramServer, TheiaSprottySelectionForwarder } from 'sprotty-theia/lib';
 import { TheiaContextMenuService } from 'sprotty-theia/lib/sprotty/theia-sprotty-context-menu-service';
 
+import { createWorkflowDiagramContainer } from '../../../../coffee-workflow-sprotty';
```
[the remark this answers is not in the document]
Comment (cdamus) [reply web/coffee-workflow-glsp-editor/src/browser/diagram/workflow-diagram-configuration.ts:29]: If that works, then yes, it's tidier. I think I emulated another import in the code. 😀

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts (hunk 1 above)
Comment (cdamus) [reply web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:2]: Absolutely. I originally touched this file late last year but only committed it now. However, it's probably older than 2020, so I should try to track down the first year for precisely.

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts (hunk 2 above)
Comment (cdamus) [reply web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:14]: Ah, OK. Not sure where this copyright came from. Probably from the **glsp-examples** project. If so, I don't know that it's so easy to change the license.

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts (hunk 1 above)
Comment (cdamus) [reply web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:2]: I'm wrong. I did not make any changes to file; it is exactly as it is in the **glsp-examples** repository. So, the copyright date would only change if the the license statement is changed in this and all of the other files.

File: web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts (hunk 2 above)
Comment (cdamus) [reply web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:14]: Confirmed. This is the license statement used in all of these files in the **glsp-examples** project. So, the question then is what is the procedure for re-licensing this? Does that need a CQ? The originating project (glsp-examples) is also managed under the Eclipse Foundation, if I'm not mistaken.

Comment (eneufeld) [reply web/coffee-workflow-sprotty/src/direct-task-editing/di.config.ts:14]: yes the other project is managed under the Eclipse Foundation, also the code was written by us. I ask Max, Jonas and Philip.

**Review by eneufeld — COMMENTED** [review]
Works for me, I double check the licencing and with the changed import we can merge this.

**Comment by eneufeld** [comment]
I created a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23142

**Comment by cdamus** [comment]
> I created a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23142

Great! Thanks It looks like the CQ is ready to be returned to the IP team. If so, we need to hit that button in their process workflow.

One other question: will we need to copy over the whole Git history of these files, also? I think it can be done with some tricky Git tree slicing and path rewriting, but I've never done it before.

**Comment by eneufeld** [comment]
> > I created a CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23142
> 
> Great! Thanks It looks like the CQ is ready to be returned to the IP team. If so, we need to hit that button in their process workflow.
> 
> One other question: will we need to copy over the whole Git history of these files, also? I think it can be done with some tricky Git tree slicing and path rewriting, but I've never done it before.

I don't see a button :-( .
I don't think we will need to keep the history, as even when doing an initial contribution the history is erased.

**Comment by eneufeld** [comment]
@cdamus the cq was approved, so I think it is fine to change the license and merge this.

**Review by eneufeld — APPROVED** [review]
Thank you very much. 
I think it makes sense to squash the 3 commits, what do you think?
You can go ahead and merge whenever it fits you.

**Comment by cdamus** [comment]
Hi @eneufeld, I've pushed commit 41306d709206ed2fbf2b290e2ae79c7852d1a8f3 that squashes the branch but I cannot review this PR myself and cannot merge it.

**Review by eneufeld — APPROVED** [review]
Thank you!

**Pull request merged**